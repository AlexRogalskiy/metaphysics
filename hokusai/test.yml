---
version: '2'
services:
  metaphysics:
    command: yarn test
    extends:
      file: common.yml
      service: metaphysics
    environment:
    - NODE_ENV=test
    env_file: ../.env.test
