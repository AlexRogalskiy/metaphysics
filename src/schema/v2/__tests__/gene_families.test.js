/* eslint-disable promise/always-return */
import { runV2Query } from "test/utils"

describe("GeneFamilies", () => {
  const api_data = [
    {
      id: "design-concepts-and-techniques",
      name: "Design Concepts and Techniques",
    },
    {
      id: "furniture-and-lighting",
      name: "Furniture & Lighting",
    },
  ]

  it("returns a list of gene families", () => {
    const geneFamiliesLoader = () => Promise.resolve(api_data)
    const query = `
      {
        gene_families {
          edges {
            node {
              id
              name
            }
          }
        }
      }
    `

    return runV2Query(query, { geneFamiliesLoader }).then(geneFamilies => {
      expect(geneFamilies).toMatchSnapshot()
    })
  })
})
