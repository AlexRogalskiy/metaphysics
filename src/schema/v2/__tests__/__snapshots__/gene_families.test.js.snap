// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneFamilies returns a list of gene families 1`] = `
Object {
  "geneFamilies": Object {
    "edges": Array [
      Object {
        "node": Object {
          "name": "Design Concepts and Techniques",
          "slug": "design-concepts-and-techniques",
        },
      },
      Object {
        "node": Object {
          "name": "Furniture & Lighting",
          "slug": "furniture-and-lighting",
        },
      },
    ],
  },
}
`;
